packages/echo-base/src/errors/BaseError.ts

```diff
@@ -146,13 +149,29 @@ export function getAllProperties(
     names.forEach((name) => {
         const value = object[name];
         const valueType = typeof value;
-        if (valueType === 'function') {
+        if (valueType === 'function' || shouldIgnore(name, args)) {
             //ignore
-        } else if (value instanceof Error) {
-            rec[name] = getAllProperties(value);
+        } else if (typeof value === 'object') {
+            rec[name] = getAllProperties(value, args);
         } else {
             rec[name] = value;
         }
     });
     return rec;
 }
+
+function shouldIgnore(name: string, args?: { ignoreEquals?: string[]; ignoreIncludes?: string[] }) {
```

Comment:
Since it returns with a `boolean`, how about renaming to follow the boolean naming patter, by adding an 'is' prefix.

maybe something like this would work?
`isPropertyIgnored()`